code/espurna/telnet.ino

```diff
@@ -143,31 +232,30 @@ void _telnetNewClient(AsyncClient *client) {
             client->close(true);
             return;
         }
-
     }
 
     for (unsigned char i = 0; i < TELNET_MAX_CLIENTS; i++) {
 
-        if (!_telnetClients[i] || !_telnetClients[i]->connected()) {
+        if (_telnetClients[i].disconnected()) {
 
-            _telnetClients[i] = client;
+            _telnetClients[i] = *client;
```

Comment:
I thinks this leaks. `client` ptr data is copied to our array but the actual pointer is never destroyed.

Just keep the capture technique that was used previously or, if you want to keep the methods looking common (i.e. dot syntax), you can use std::unique_ptr as AsyncClient pointer holder and [`.reset(new_ptr)`](https://en.cppreference.com/w/cpp/memory/unique_ptr/reset) when needed. ...ptr can be nulled, which deletes pointer. Bool checks still work (if (unique_ptr....)), checking if captured pointer is present.

Reply:
Similarly, `new WiFiClient(...)` can be used. Since destructor stops it, maybe disconnect part can be simplified to just setting unique_ptr list member to null?

Reply:
Hmm, something like this? ``_telnetClients[i] = std::unique_ptr<AsyncClient>(client);``
We can't use the dot syntax then though, so we'd have to change the WiFiClient array to ``std::unique_ptr`` as well (correct me if I'm wrong).

Reply:
Nah, the std::unique_ptr can always be there. Array declaration like that does object construction asap, there are TELNET_MAX_CLIENTS WiFiClient objects created on boot.
```
WiFiClient _telnetClients[TELNET_MAX_CLIENTS];
...
std::unique_ptr<TELNET_CLIENT_CLASS> _telnetClients[TELNET_MAX_CLIENTS];
_telnetClients[0] = some_pointer_obj;
_telnetClients[0]->write(...);
```

Yes, you right, no dot syntax. ->

Reply:
Thanks for the pointers! (sic)

I have updated the PR, it compiles fine but I'm unable to check right now because my dev board seems to have died on me.

Reply:
Updated with fixes for the last issues - checked and I can confirm both server implementations are working fine now.